wsd/RequestVettingStation.cpp

```diff
@@ -271,29 +272,21 @@ void RequestVettingStation::handleRequest(const std::string& id,
                                                                            << "] in config");
             sendUnauthorizedErrorAndShutdown();
             break;
-
 #if ENABLE_LOCAL_FILESYSTEM
         case StorageBase::StorageType::FileSystem:
             LOG_INF("URI [" << COOLWSD::anonymizeUrl(uriPublic.toString()) << "] on docKey ["
                             << docKey << "] is for a FileSystem document");
 
-            // Remove from the current poll and transfer.
-            disposition.setMove(
-                [selfLifecycle = shared_from_this(), this, docKey = std::move(docKey),
-                 url = std::move(url), uriPublic](const std::shared_ptr<Socket>& moveSocket)
-                {
-                    LOG_TRC_S('#' << moveSocket->getFD()
-                                  << ": Dissociating client socket from "
-                                     "ClientRequestDispatcher and creating DocBroker for ["
-                                  << docKey << ']');
-
-                    // Create the DocBroker.
-                    if (std::shared_ptr<DocumentBroker> docBroker = createDocBroker(docKey, "",
-                                url, uriPublic))
-                    {
-                        createClientSession(docBroker, docKey, url, uriPublic);
-                    }
-                });
+            LOG_TRC("Dissociating client socket from "
+                             "ClientRequestDispatcher and creating DocBroker for ["
+                          << docKey << ']');
+
+            // Create the DocBroker.
+            if (std::shared_ptr<DocumentBroker> docBroker = createDocBroker(docKey, "",
+                        url, uriPublic))
+            {
+                createClientSession(docBroker, docKey, url, uriPublic);
```

Comment:
Not so happy about the createClientSession not happening in a callback inside the docBroker's poll ;-) But I imagine this is lowish-risk and unlikely to cause grief.

Reply:
I think I have a piece like this that I can do in follow up